custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/impl/plugins/UpdateCenterPluginInfoProvider.java

```diff
@@ -0,0 +1,61 @@
+package io.jenkins.tools.warpackager.lib.impl.plugins;
+
+import io.jenkins.tools.warpackager.lib.config.DependencyInfo;
+import io.jenkins.tools.warpackager.lib.model.plugins.PluginInfoProvider;
+import net.sf.json.JSONObject;
+import org.apache.commons.io.IOUtils;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class UpdateCenterPluginInfoProvider implements PluginInfoProvider {
+
+    public static final String DEFAULT_JENKINS_UC_URL = "http://updates.jenkins.io/update-center.json";
+    private static final Logger LOGGER = Logger.getLogger(UpdateCenterPluginInfoProvider.class.getName());
+
+    private String updateCenterUrl;
+    private Map<String, String> groupIdCache;
+
+    public UpdateCenterPluginInfoProvider(@Nonnull String updateCenterUrl) {
+        this.updateCenterUrl = updateCenterUrl;
+    }
+
+    @Override
+    public void init() throws IOException, InterruptedException {
+        groupIdCache = extractUpdateCenterData(new URL(updateCenterUrl));
+    }
+
+    @Override
+    public boolean isPlugin(@Nonnull DependencyInfo dependency) throws IOException, InterruptedException {
+        LOGGER.log(Level.FINE, "Checking whether {0} is a plugin: {1}",
+                new Object[] {dependency.artifactId, groupIdCache.containsKey(dependency.artifactId)});
+        return groupIdCache.containsKey(dependency.artifactId);
+    }
+
+    private static Map<String, String> extractUpdateCenterData(URL url) throws IOException {
+        Map<String, String> groupIDs = new HashMap<>();
+        String jsonp = null;
+        try {
+            jsonp = IOUtils.toString(url.openStream());
+        } catch(IOException e){
+            throw new IOException("Invalid update center url : " + url, e);
+        }
+
+        String json = jsonp.substring(jsonp.indexOf('(')+1,jsonp.lastIndexOf(')'));
+        JSONObject jsonObj = JSONObject.fromObject(json);
+
+        // UpdateSite.Plugin does not contain gav object, so we process the JSON object on our own here
+        for(Map.Entry<String,JSONObject> e : (Set<Map.Entry<String,JSONObject>>)jsonObj.getJSONObject("plugins").entrySet()) {
```

Comment:
P2: A bit of refactoring to extract a few unit-testable methods for this parsing would help us write unit tests that would hopefully help me understand better/faster what's going on here and what the key and value pairs are.

After formatting the JSON I got from the update center, it looks like, given a fragment like:
```
        "AnchorChain": {
            "buildDate": "Mar 11, 2012",
(...)
            "gav": "org.jenkins-ci.plugins:AnchorChain:1.0",
(...)
            "wiki": "https://plugins.jenkins.io/AnchorChain"
        },
```

...the key is `AnchorChain`, which appears to be the extracted from the artifactId portion of the `gav` value, and thus we are creating a lookup of plugins by their artifactId and storing their groupId as the associated value.

If I understood that right, I would like to make a suggestion: instead of creating a `HashMap<>`, consider creating a `TreeMap<>(String.CASE_INSENSITIVE_ORDER)`.  This will impact performance somewhat (`O(log(n))` instead of `O(1)`) but then users won't get "plugin not found" if they don't have the exact casing of the artifactId, as, until now, I thought all plugins had lowercased artifactId. (assuming, of course, there don't exist distinct plugins whose names differ only by case)

Reply:
> If I understood that right, I would like to make a suggestion: instead of creating a HashMap<>, consider creating a TreeMap<>(String.CASE_INSENSITIVE_ORDER). This will impact performance somewhat (O(log(n)) instead of O(1)) but then users won't get "plugin not found" if they don't have the exact casing of the artifactId, as, until now, I thought all plugins had lowercased artifactId. (assuming, of course, there don't exist distinct plugins whose names differ only by case)

I prefer not to touch it, because it may lead to artifact ID resolution issues in some cases. I also plan to use https://github.com/jenkinsci/plugin-installation-manager-tool in this repo eventually, and I would prefer not to touch the case sensitivity until there is a support for it in the underlying tool. CC @stopalopa , It might be an interesting RFE for your project